core/src/main/java/org/elasticsearch/index/translog/MultiSnapshot.java

```diff
@@ -41,18 +36,7 @@
      */
     MultiSnapshot(Translog.Snapshot[] translogs) {
         this.translogs = translogs;
-        int ops = 0;
-        for (Translog.Snapshot translog : translogs) {
-
-            final int tops = translog.estimatedTotalOperations();
-            if (tops == TranslogReader.UNKNOWN_OP_COUNT) {
-                ops = TranslogReader.UNKNOWN_OP_COUNT;
-                break;
-            }
-            assert tops >= 0 : "tops must be positive but was: " + tops;
-            ops += tops;
-        }
-        estimatedTotalOperations = ops;
+        estimatedTotalOperations = Arrays.stream(translogs).mapToInt(Translog.Snapshot::estimatedTotalOperations).sum();
```

Comment:
Yeah, but if memory serves it was either “I don’t know” or “I do know”. Anyway, long gone happily.

> On 12 Jan 2016, at 10:07, Simon Willnauer <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/index/translog/MultiSnapshot.java:
> 
> > @@ -41,18 +36,7 @@
> >       */
> >      MultiSnapshot(Translog.Snapshot[] translogs) {
> >          this.translogs = translogs;
> > -        int ops = 0;
> > -        for (Translog.Snapshot translog : translogs) {
> > -
> > -            final int tops = translog.estimatedTotalOperations();
> > -            if (tops == TranslogReader.UNKNOWN_OP_COUNT) {
> > -                ops = TranslogReader.UNKNOWN_OP_COUNT;
> > -                break;
> > -            }
> > -            assert tops >= 0 : "tops must be positive but was: " + tops;
> > -            ops += tops;
> > -        }
> > -        estimatedTotalOperations = ops;
> > +        estimatedTotalOperations = Arrays.stream(translogs).mapToInt(Translog.Snapshot::estimatedTotalOperations).sum();
> 
> it was when we had to read until EOF to figure out how many :)
> 
> —
> Reply to this email directly or view it on GitHub.
>